src/components/AssetPackEditor/AssetPackEditor.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react'
+import { Field, Header, Button } from 'decentraland-ui'
+import AssetThumbnail from 'components/AssetThumbnail'
+import { RawAssetPack } from 'modules/assetPack/types'
+import { isGround } from 'modules/asset/utils'
+import Icon from 'components/Icon'
+
+import { Props, State } from './AssetPackEditor.types'
+import './AssetPackEditor.css'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { MAX_NAME_LENGTH, MAX_THUMBNAIL_SIZE } from 'modules/assetPack/utils'
+
+export default class AseetPackEditor<T extends RawAssetPack> extends React.PureComponent<Props<T>, State> {
+  thumbnailInput = React.createRef<HTMLInputElement>()
+
+  state: State = {
+    errors: {}
+  }
+
+  componentDidMount() {
+    const { assetPack, onChange } = this.props
+
+    if (!assetPack.thumbnail) {
+      const asset = assetPack.assets.length ? assetPack.assets.find(asset => !isGround(asset)) : null
+      onChange({
+        ...assetPack,
+        thumbnail: asset ? asset.thumbnail : ''
+      })
+    }
+  }
+
+  handleRemove = (id: string) => {
+    const { assetPack, onChange } = this.props
+
+    onChange({
+      ...assetPack,
+      assets: assetPack.assets.filter(asset => asset.id !== id)
+    })
+  }
+
+  handleOpenFileDialog = () => {
+    if (this.thumbnailInput.current) {
+      this.thumbnailInput.current.click()
+    }
+  }
+
+  handleThumbnail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { assetPack, onChange } = this.props
+    const { files } = e.target
+
+    if (files) {
+      const file = files[0]
+      if (file.size > MAX_THUMBNAIL_SIZE) {
```

Comment:
II got an exeption on this line because `file` was `undefined`. I don't know how to reproduce, but that `if (files) { ` above is not stopping an empty array from going through:

```
const files = []
if (files) {
 console.log('boom') // 💥
}
```

We should make it `if (files.length > 0) {`